### PR TITLE
Dashboard: Change cursor sync (graphTooltip) to shared cursor

### PR DESCRIPTION
**Pull request by torkelo**
* [x] Changes default for "Graph tooltip" from Off to "Shared crosshair" 
* [x] Renames settings label from "Graph tooltip" to "Cursor sync"  (very hard to name this option). 
* [x] Updates description to "Controls tooltip and hover highlight behavior across different panels. Reload the dashboard for changes to take effect" to "Controls the synchronized hover effect across different panels. Reload the dashboard for changes to take effect" 
* [x] Changes the Off text label from "Default" to "Off"

### CODE REVIEW

**Comment by leeoniya** [comment]
we're in the process of removing Y sync. not sure if this PR should wait until that's done to avoid changing the new default behavior?

also have some concerns about performance of this in large dashboards. but probably okay?

**Review by dprokop — APPROVED** [review]
Looking good. 

Just a nit to consider is the way the new defautls (or rather migrations) are applied. I mean - If I had dashboards already created and the sync was suddenly ON for all of them after the migration I may consider this a "regression". So maybe we should enable this new default behavior for newly created dashbaords only?

File: public/app/features/dashboard/state/DashboardModel.ts [hunk 1]
```diff
@@ -144,7 +143,7 @@ export class DashboardModel implements TimeModel {
     this.timezone = data.timezone ?? '';
     this.weekStart = data.weekStart ?? '';
     this.editable = data.editable !== false;
-    this.graphTooltip = data.graphTooltip || 0;
+    this.graphTooltip = data.graphTooltip || DashboardCursorSync.Crosshair;
```
[the remark this answers is not in the document]
Comment (torkelo) [reply public/app/features/dashboard/state/DashboardModel.ts:146]: @dprokop ah, that is a bug, did not notice the use of `||` there, should be `??`

**Comment by leeoniya** [comment]
still landing sync-related changes, like optimizations to EventBus publish (which is active with cursor sync): https://github.com/grafana/grafana/pull/84111

**Comment by leeoniya** [comment]
i'm not sure the new default is a good one.

sync is not free, and in busy dashboards which can lazy-load hundreds of panels and thousands of series, this becomes a rather annoying UX perf problem since sync does not currently take into account whether a synced panel is within the visible viewport.

renaming it is good tbough.

**Pull request closed without merging**